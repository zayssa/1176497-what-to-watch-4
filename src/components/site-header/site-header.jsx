import React from "react";
import PropTypes from "prop-types";
import {Link} from "react-router-dom";
import {connect} from "react-redux";
import {getAuthStatus} from "../../reducer/selectors";
import {IUser} from "../../types/user";


const SiteHeader = ({authorizationStatus, userInfo}) => {
  return (
    <>
      <h1 className="visually-hidden">WTW</h1>

      <header className="page-header movie-card__head">
        <div className="logo">
          <Link to="/" className="logo__link">
            <span className="logo__letter logo__letter--1">W</span>
            <span className="logo__letter logo__letter--2">T</span>
            <span className="logo__letter logo__letter--3">W</span>
          </Link>
        </div>

        <div className="user-block">
          {authorizationStatus === `NO_AUTH` ? (
            <Link to="/login" className="user-block__link">Sign in</Link>
          ) : (
            <div className="user-block__avatar">
              <Link to="/mylist">
                <img src={`https://4.react.pages.academy${userInfo.avatar_url}`} alt="User avatar" width="63" height="63" />
              </Link>
            </div>
          )}
        </div>
      </header>
    </>
  );
};

const mapStateToProps = (state) => ({
  authorizationStatus: getAuthStatus(state)
});

const mapDispatchToProps = () => ({});

SiteHeader.propTypes = {
  authorizationStatus: PropTypes.string.isRequired,
  userInfo: IUser
};

export {SiteHeader};
export default connect(mapStateToProps, mapDispatchToProps)(SiteHeader);
