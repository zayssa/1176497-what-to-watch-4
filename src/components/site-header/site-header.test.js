import React from "react";
import {Router} from "react-router-dom";
import renderer from "react-test-renderer";
import {SiteHeader} from "./site-header.jsx";
import history from "../../history";


it(`Render SiteHeader`, () => {
  const tree = renderer
    .create(<Router history={history}><SiteHeader authorizationStatus="NO_AUTH" /></Router>)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
