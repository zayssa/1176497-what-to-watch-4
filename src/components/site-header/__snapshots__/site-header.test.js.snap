// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render SiteHeader 1`] = `
Array [
  <h1
    className="visually-hidden"
  >
    WTW
  </h1>,
  <header
    className="page-header movie-card__head"
  >
    <div
      className="logo"
    >
      <a
        className="logo__link"
        href="/"
        onClick={[Function]}
      >
        <span
          className="logo__letter logo__letter--1"
        >
          W
        </span>
        <span
          className="logo__letter logo__letter--2"
        >
          T
        </span>
        <span
          className="logo__letter logo__letter--3"
        >
          W
        </span>
      </a>
    </div>
    <div
      className="user-block"
    >
      <a
        className="user-block__link"
        href="/login"
        onClick={[Function]}
      >
        Sign in
      </a>
    </div>
  </header>,
]
`;
